kinto/core/storage/postgresql/schema.sql

```diff
@@ -75,18 +75,20 @@ RETURNS TIMESTAMP AS $$
 DECLARE
     ts TIMESTAMP;
 BEGIN
-    ts := NULL;
-
-    SELECT last_modified INTO ts
-      FROM timestamps
-     WHERE parent_id = uid
-       AND collection_id = resource;
-
-    IF ts IS NULL THEN
-      ts := clock_timestamp();
+    WITH create_if_missing AS (
       INSERT INTO timestamps (parent_id, collection_id, last_modified)
-      VALUES (uid, resource, ts);
-    END IF;
+      VALUES (uid, resource, clock_timestamp())
+      ON CONFLICT (parent_id, collection_id) DO NOTHING
+      RETURNING last_modified
```

Comment:
Why not? If it's inserted it will return the inserted value, if there's a conflict it won't return anything. That's why I did an `UNION` after to do like *create or get*